From 286a4e96a306390ea56dcf639eec6e86443de24c Mon Sep 17 00:00:00 2001
From: Martin Pitt <martin.pitt@ubuntu.com>
Date: Wed, 30 Apr 2014 15:57:30 +0200
Subject: [PATCH] libmm-glib: Initialize result array

In mm_modem_messaging_get_supported_storages(), initialize array to avoid a
potential crash with an uninitialized value, and a build failure due to

mm-modem-messaging.c:168:24: error: 'array' may be used uninitialized in this
function [-Werror=maybe-uninitialized]
     *n_storages = array->len;

Bug: https://bugzilla.gnome.org/show_bug.cgi?id=729267
---
 libmm-glib/mm-modem-messaging.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libmm-glib/mm-modem-messaging.c b/libmm-glib/mm-modem-messaging.c
index fdf1497..c619194 100644
--- a/libmm-glib/mm-modem-messaging.c
+++ b/libmm-glib/mm-modem-messaging.c
@@ -155,7 +155,7 @@ mm_modem_messaging_get_supported_storages (MMModemMessaging *self,
                                            MMSmsStorage **storages,
                                            guint *n_storages)
 {
-    GArray *array;
+    GArray *array = NULL;
 
     g_return_val_if_fail (MM_IS_MODEM_MESSAGING (self), FALSE);
     g_return_val_if_fail (storages != NULL, FALSE);
-- 
2.0.0.rc0

